ignite/chainconfig/parser.go

```diff
@@ -0,0 +1,146 @@
+package chainconfig
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/ignite/cli/ignite/chainconfig/config"
+	v1 "github.com/ignite/cli/ignite/chainconfig/v1"
+)
+
+// Parse reads a config file.
+// When the version of the file beign read is not the latest
+// it is automatically migrated to the latest version.
+func Parse(configFile io.Reader) (*v1.Config, error) {
+	var buf bytes.Buffer
+
+	// Read the config file version first to know how to decode it
+	version, err := ReadConfigVersion(io.TeeReader(configFile, &buf))
+	if err != nil {
+		return DefaultConfig(), err
```

Comment:
Why no returning `nil` here instead of `DefaultConfig()` ?

Reply:
The original implementation was returning the struct already so I kept the behaviour. But I believe the "benefit" is that the config is potentially not kept in the heap memory if a non pointer is returned, so it doesn't need to be garbage collected. Also, arguably :), returning a non pointer might be semantically correct to imply that the intention of the returned config is only to be read from this point on.